File: Java/RecyclerView12RecyclerViewToolbarScrollEffect/app/src/main/java/dev/ogabek/recyclerview12_recyclerviewtoolbarscrolleffect/activity/MainActivity.java
package dev.ogabek.recyclerview12_recyclerviewtoolbarscrolleffect.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.os.Bundle;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.recyclerview12_recyclerviewtoolbarscrolleffect.R;
import dev.ogabek.recyclerview12_recyclerviewtoolbarscrolleffect.adapter.MainRecyclerViewAdapter;
import dev.ogabek.recyclerview12_recyclerviewtoolbarscrolleffect.model.Member;

public class MainActivity extends AppCompatActivity {

    Context context;
    RecyclerView recyclerView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        refreshAdapter(prepareMemberList());

    }

    private List<Member> prepareMemberList() {
        List<Member> members = new ArrayList<>();
        for (int i = 0; i < 50; i++) members.add(new Member());
        return members;
    }

    private void initViews() {
        context = this;
        recyclerView = findViewById(R.id.rv_mainRecyclerView);
        recyclerView.setLayoutManager(new GridLayoutManager(context, 1));
    }

    private void refreshAdapter(List<Member> members) {
        MainRecyclerViewAdapter adapter = new MainRecyclerViewAdapter(context, members);
        recyclerView.setAdapter(adapter);
    }

}

File: Kotlin/RecyclerView1RecyclerViewCustomBasicAdapter/app/src/main/java/dev/ogabek/recyclerview1_recyclerviewcustombasicadapter/activity/MainActivity.kt
package dev.ogabek.recyclerview1_recyclerviewcustombasicadapter.activity

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.recyclerview1_recyclerviewcustombasicadapter.R
import dev.ogabek.recyclerview1_recyclerviewcustombasicadapter.adapter.MainAdapter
import dev.ogabek.recyclerview1_recyclerviewcustombasicadapter.model.Member

class MainActivity : AppCompatActivity() {

    private lateinit var recyclerView: RecyclerView
    private lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

        refreshAdapter(prepareMemberList())

    }

    private fun initViews() {
        context = this
        recyclerView = findViewById(R.id.rv_main_recycler_view)
        recyclerView.layoutManager = GridLayoutManager(context, 1)
    }

    private fun refreshAdapter(members: List<Member>) {
        val adapter = MainAdapter(context, members)
        recyclerView.adapter = adapter
    }

    private fun prepareMemberList(): List<Member> {
        val members = ArrayList<Member>()
        for (i in 0..15) {
            members.add(Member(R.drawable.img, "Ogabek Matyakubov", "Mobile Developer"))
        }
        return members
    }

}